src/data/set/function.lean

```diff
@@ -331,6 +331,9 @@ begin
   { simp [nat.iterate, ihn] }
 end
 
+lemma maps_to_of_subsingleton [subsingleton α] (f : α → α) (s : set α) : maps_to f s s :=
```

Comment:
Is this true for two distinct subsingleton types?

Reply:
No, because you need the set to be the same on both sides. `maps_to f {a} ∅` is definitely false.